packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -26,38 +38,32 @@ export function useDebouncedShowMovers( {
 		return ! isFocused && ! isHovered;
 	};
 
-	const debouncedShowMovers = useCallback(
-		( event ) => {
-			if ( event ) {
-				event.stopPropagation();
-			}
+	const debouncedShowMovers = ( event ) => {
+		if ( event ) {
+			event.stopPropagation();
+		}
 
-			const timeout = timeoutRef.current;
+		const timeout = timeoutRef.current;
 
-			if ( timeout && clearTimeout ) {
-				clearTimeout( timeout );
-			}
-			if ( ! showMovers ) {
-				setShowMovers( true );
-			}
-		},
-		[ showMovers ]
-	);
+		if ( timeout && clearTimeout ) {
+			clearTimeout( timeout );
+		}
+		if ( ! showMovers ) {
+			handleOnChange( true );
+		}
+	};
 
-	const debouncedHideMovers = useCallback(
-		( event ) => {
-			if ( event ) {
-				event.stopPropagation();
-			}
+	const debouncedHideMovers = ( event ) => {
+		if ( event ) {
+			event.stopPropagation();
+		}
 
-			timeoutRef.current = setTimeout( () => {
-				if ( shouldHideMovers() ) {
-					setShowMovers( false );
-				}
-			}, debounceTimeout );
-		},
-		[ isFocused ]
-	);
+		timeoutRef.current = setTimeout( () => {
```

Comment:
It existed previously, but: Is there any chance that this function is called twice before the callback is invoked? Because it could cause a problem where the `timeoutRef.current` is assigned to whichever `setTimeout` was scheduled last, and only the last one would be unscheduled in the `useEffect` unsubscribe callback (i.e. the first one would still trigger its callback even after the component unmounts, causing a warning to be logged if it calls to set state).

Reply:
Oh! Hmm... Perhaps we should clear the timeout before setting it in this debounce function.